include/cantera/thermo/Phase.h

```diff
@@ -785,6 +804,26 @@ class Phase
     //! Entropy at 298.15 K and 1 bar of stable state pure elements (J kmol-1)
     vector_fp m_entropy298;
 
+
+// added
+public:
+        std::vector<std::string> spPos;
+        std::vector<int> spPosIndex;
+        std::vector<std::string> spNeg;
+        std::vector<int> spNegIndex;
+        std::vector<std::string> spNeut;
+        std::vector<int> spNeutIndex;
+
+
+        std::vector<int> indexNeutNeut;
+        std::vector<int> indexNeutPos;
+        std::vector<int> indexNeutNeg;
+        std::vector<int> indexPosPos;
+        std::vector<int> indexPosNeg;
+        std::vector<int> indexNegNeg;
```

Comment:
For most classes in Cantera, the convention is that member variables are prefixed with "m_", e.g. "m_indexPosPos". There need to be comments describing what these variables are.

Most member variables in class Phase are "protected" not "public". I don't immediately see a reason that these need to be public.

I'm not sure why member variables for the names of the species of each type are needed. If you have the list of species indices, you can always retrieve the name efficiently.